src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -473,6 +522,18 @@ public int countApplicationUpdatedMessages() {
      *                                  failed.
      */
     public void startServer() throws PluginExecutionException {
+        if (container) {
+            File dockerfileToUse = dockerfile != null ? dockerfile : defaultDockerfile;
+            debug("Dockerfile to use: " + dockerfileToUse);
+            if (dockerfileToUse.exists()) {
+                List<String> dockerfileLines = removeCopyLines(removeWarFileLines(readDockerfile(dockerfileToUse)), dockerfileToUse.getParent());
```

Comment:
In general, nesting method calls like this makes support difficult. Why not let the createTempDockerfile method call each of those sequentially and return the temp dockerfile. Maybe call it prepareTempDockerfile to indicate it does some processing before creating it.

Reply:
When using the defaultDockerfile, I think calling removeWarFileLines and removeCopyLines is unnecessary isn't it? And how exactly would the defaultDockerfile not exist?

Reply:
defaultDockerfile is not a default dockerfile that we create.  Instead, it just represents the default dockerfile location that we are expecting.  It doesn't exist if the user does not have a Dockerfile in their project.  So regardless, we still need to call  removeWarFileLines and removeCopyLines since the Dockerfile is a user-provided file.

Reply:
Changed to prepareTempDockerfile and sequential calls from there